contracts/V1/StateChannelDiamondProxy/StateChannelUtilLibrary.sol

```diff
@@ -90,33 +90,57 @@ library StateChannelUtilLibrary {
     function tryInsertAddressInThresholdSet(
         address adr,
         address[] memory set,
-        uint currentSetLength,
+        uint currentThresholdCount,
         address[] memory expectedAddresses
     ) internal pure returns (uint) {
-        bool found = false;
         //Check is address in expectedAddresses
         for (uint i = 0; i < expectedAddresses.length; i++) {
             if (expectedAddresses[i] == adr) {
-                found = true;
-                break;
+                if(set[i] != adr) {
+                    set[i] = adr;
+                    currentThresholdCount++;
+                    break;
+                }
             }
         }
-        if (!found) return currentSetLength;
-        //Try and insert
-        for (uint i = 0; i < currentSetLength; i++) {
-            if (set[i] == adr) return currentSetLength;
-        }
-        set[currentSetLength] = adr;
-        return currentSetLength + 1;
+        return currentThresholdCount;
     }
 
     /// @dev Concatenates two address arrays, but it does not add duplicates
-    function concatAddressArrays(address[] memory array1, address[] memory array2, uint array1Length) internal pure returns (address[] memory) {
-       address[] memory result = new address[](array1Length);
-       
-       for(uint i = 0; i < array2.length; i++) {
-        result[i] = array2[i];
-       }
+    function concatAddressArrays(address[] memory array1, address[] memory array2) internal pure returns (address[] memory) {
+        address[] memory result = new address[](array1.length + array2.length);
+        for (uint i = 0; i < array1.length; i++) {
+            result[i] = array1[i];
+        }
+        for (uint i = 0; i < array2.length; i++) {
+            result[array1.length + i] = array2[i];
+        }
        return result;
     }
+
+    function concatExitChannelArrays(ExitChannel[] memory array1, ExitChannel[] memory array2) internal pure returns (ExitChannel[] memory) {
+        ExitChannel[] memory result = new ExitChannel[](array1.length + array2.length);
+        for (uint i = 0; i < array1.length; i++) {
+            result[i] = array1[i];
+        }
+        for (uint i = 0; i < array2.length; i++) {
+            result[array1.length + i] = array2[i];
+        }
+       return result;
+    }
+
+    function areAddressArraysEqual(
```

Comment:
can you estimate the typical size of arrays?

maybe hash based approach could be more efficient
```solidity
function areAddressArraysEqual(
    address[] memory array1,
    address[] memory array2
) internal pure returns (bool) {
    if (array1.length != array2.length) {
        return false;
    }
    
    bytes32 hash1 = keccak256(abi.encode(array1));
    bytes32 hash2 = keccak256(abi.encode(array2));
    
    return hash1 == hash2;
}
```

Reply:
With hash you still have to hash the entire array even though it's implicit and hashing is more expensive than comparison, but either way this is only used in one place for now, to compare on-chain slashed, so at most N where N is the number of players (so it's small) - we have other places that are less efficient and consume more gas that we can optimize later